freqtrade/exchange/binance.py

```diff
@@ -66,29 +87,97 @@ def stoploss(self, pair: str, amount: float, stop_price: float, order_types: Dic
             self._log_exchange_response('create_stoploss_order', order)
             return order
         except ccxt.InsufficientFunds as e:
             raise InsufficientFundsError(
-                f'Insufficient funds to create {ordertype} sell order on market {pair}. '
-                f'Tried to sell amount {amount} at rate {rate}. '
+                f'Insufficient funds to create {ordertype} {side} order on market {pair}. '
+                f'Tried to {side} amount {amount} at rate {rate}. '
                 f'Message: {e}') from e
         except ccxt.InvalidOrder as e:
             # Errors:
             # `binance Order would trigger immediately.`
             raise InvalidOrderException(
-                f'Could not create {ordertype} sell order on market {pair}. '
-                f'Tried to sell amount {amount} at rate {rate}. '
+                f'Could not create {ordertype} {side} order on market {pair}. '
+                f'Tried to {side} amount {amount} at rate {rate}. '
                 f'Message: {e}') from e
         except ccxt.DDoSProtection as e:
             raise DDosProtection(e) from e
         except (ccxt.NetworkError, ccxt.ExchangeError) as e:
             raise TemporaryError(
-                f'Could not place sell order due to {e.__class__.__name__}. Message: {e}') from e
+                f'Could not place {side} order due to {e.__class__.__name__}. Message: {e}') from e
+        except ccxt.BaseError as e:
+            raise OperationalException(e) from e
+
+    @retrier
+    def fill_leverage_brackets(self):
+        """
+            Assigns property _leverage_brackets to a dictionary of information about the leverage
+            allowed on each pair
+        """
+        try:
+            leverage_brackets = self._api.load_leverage_brackets()
+            for pair, brackets in leverage_brackets.items():
+                self._leverage_brackets[pair] = [
+                    [
+                        min_amount,
+                        float(margin_req)
+                    ] for [
+                        min_amount,
+                        margin_req
+                    ] in brackets
+                ]
+
+        except ccxt.DDoSProtection as e:
+            raise DDosProtection(e) from e
+        except (ccxt.NetworkError, ccxt.ExchangeError) as e:
+            raise TemporaryError(f'Could not fetch leverage amounts due to'
+                                 f'{e.__class__.__name__}. Message: {e}') from e
+        except ccxt.BaseError as e:
+            raise OperationalException(e) from e
+
+    def get_max_leverage(self, pair: Optional[str], nominal_value: Optional[float]) -> float:
+        """
+            Returns the maximum leverage that a pair can be traded at
+            :param pair: The base/quote currency pair being traded
+            :nominal_value: The total value of the trade in quote currency (collateral + debt)
+        """
+        pair_brackets = self._leverage_brackets[pair]
```

Comment:
> i do have my concerns about this (assuming the dict is filled for every pair) - as also stated in the kraken class already.

My original intention was to not fill it for every pair, so you could check if its in there to see if it can trade on margin/futures. With binance if it's not in there, then we'll have to deal with transfers to spot out of margin. With kraken that shouldn't be a problem, because a min of 1 is set for every pair

Reply:
i've not got a problem with that per se - but it'll mean that you must have a check for "is in this dict" either before calling this method (which is probably clumsy / might require changes in multiple areas) - or have proper fallbacks in place within this method - so it simply returns a "non-leverage" value.

Reply:
I think I'll leave it as is. Binance has it's own wallets for margin and futures, so if a pair isn't in the array, then you can't trade with it anyway, unless you do a transfer

Reply:
you can resolve this if you want, I see no problem with it